packages/admin/admin/src/table/TableBodyRow.tsx

```diff
@@ -1,40 +1,56 @@
-import { ComponentsOverrides, TableRow, Theme } from "@mui/material";
+import { ComponentsOverrides, TableRow } from "@mui/material";
+import { css, styled, Theme, useThemeProps } from "@mui/material/styles";
 import { TableRowProps } from "@mui/material/TableRow";
-import { createStyles, WithStyles, withStyles } from "@mui/styles";
-import * as React from "react";
+import { ThemedComponentBaseProps } from "helpers/ThemedComponentBaseProps";
+import React from "react";
 
 /**
  * @deprecated Use MUI X Data Grid in combination with `useDataGridRemote` instead.
  */
 export type TableBodyRowClassKey = "root" | "even" | "odd";
 
+type OwnerState = Pick<TableBodyRowProps, "index" | "hideTableHead">;
+
+const Root = styled(TableRow, {
+    name: "CometAdminTableBodyRow",
+    slot: "root",
+    overridesResolver({ ownerState }: { ownerState: OwnerState }, styles) {
+        const isOdd = ((ownerState.index || 0) + (ownerState.hideTableHead ? 1 : 0)) % 2 === 1;
+        return [styles.root, !isOdd && styles.even, isOdd && styles.odd];
+    },
+})<{ ownerState: OwnerState }>(({ ownerState: { index, hideTableHead } }) => {
+    const isOdd = ((index || 0) + (hideTableHead ? 1 : 0)) % 2 === 1;
+    if (isOdd) {
+        return css``;
+    }
+    return css``;
+});
```

Comment:
The logic here is not necessary, if nothing is returned. 

```suggestion
})();
```